src/main/kotlin/com/jetbrains/snakecharm/codeInsight/SnakemakeSectionsInShellReferenceContributor.kt

```diff
@@ -86,17 +91,38 @@ class SnakemakeSectionsInShellReferenceContributor : PsiReferenceContributor() {
     private fun addSectionReferences(
             element: PyStringLiteralExpression,
             sectionMatcher: Matcher,
-            sectionReferences: MutableList<PsiReference>
+            sectionReferences: MutableList<PsiReference>,
+            toplevelVariables: List<PyTargetExpression>
     ) {
         while (sectionMatcher.find()) {
             val sectionName = sectionMatcher.group(1)
+                    ?.replace(CompletionInitializationContext.DUMMY_IDENTIFIER_TRIMMED, "")
             val argumentName = sectionMatcher.group(3)
-            if (sectionName.isEmpty() || ALLOWED_IN_SHELL_WITHOUT_KEYWORDS.any { it.startsWith(sectionName) }) {
+
+            if (sectionName == null) {
+                continue
+            }
+
+            val variable = toplevelVariables.find { it.name == sectionName }
+            if (variable != null) {
```

Comment:
Are you sure in the order of candidates for resolve? Top-level vars has the highest priority?